### PR TITLE
Rolling period

### PR DESCRIPTION
**Pull request by AliceBilc**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn -P integration-test clean verify` runs for the whole project and, if you touched any code for a service in the `services` folder, ensure it can be run with `spring-boot:run`

## Description

Check rolling period in order to have valid Diagnosis Keys

### CODE REVIEW

File: common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java [hunk 1]
```diff
@@ -1,13 +1,14 @@
 package app.coronawarn.server.common.persistence.repository;
 
+import static org.junit.Assert.assertEquals;
```
Comment (FelixRottler) [inline common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java:3]: The assert statements are coming from junit4, please use the ones that are coming from junit5 (org.junit.jupiter)

File: common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java [hunk 2]
```diff
@@ -32,23 +33,46 @@ void tearDown() {
...
+        key.getReportType().name(), key.getDaysSinceOnsetOfSymptoms(),
+        key.isConsentToFederation(), key.getSubmissionType().name());
 
     assertTrue(repository.exists(id, type.name()));
   }
+
+  @Test
+  void shouldNotAddAnInvalidDiagnosisKey() {
+
+    Exception exception = assertThrows(InvalidDiagnosisKeyException.class, () -> {
```
Comment (FelixRottler) [inline common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java:63]: id will only contain 0's in this test, is this wanted?

File: common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java [hunk 3]
```diff
@@ -32,23 +33,46 @@ void tearDown() {
...
+      DiagnosisKey.builder()
```
Comment (FelixRottler) [inline common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java:64]: I guess this test would be better suited inside DiagnosisKeyBuilderTest.

File: services/submission/src/main/resources/application.yaml [hunk 4]
```diff
@@ -39,6 +39,8 @@ services:
     maximum-request-size: ${MAXIMUM_REQUEST_SIZE:100KB}
     # The maximum rolling period accepted by the SubmissionController (e.g. 144 (24 hours))
     max-rolling-period: ${MAX_ROLLING_PERIOD:144}
+    # The minimum rolling period accepted by the SubmissionController (e.g. 1 (24 hours))
```
Comment (FelixRottler) [inline services/submission/src/main/resources/application.yaml:42]: I think the comment contains an error. A rolling period of 1 does not equal 24 hours but 10 minutes

File: services/submission/src/test/java/app/coronawarn/server/services/submission/integration/SubmissionPersistenceIT.java [hunk 5]
```diff
@@ -85,6 +97,33 @@ public void setUpMocks() {
...
+    headers.setContentType(MediaType.valueOf("Application/x-protobuf"));
+
+    SubmissionPayload validSubmissionPayload = buildSubmissionPayload(List.of("DE"), "DE", true,
+        validKeys, SubmissionType.SUBMISSION_TYPE_PCR_TEST);
+
+    testRestTemplate.postForEntity("/version/v1/diagnosis-keys", new HttpEntity<>(invalidSubmissionPayload, headers), Void.class);
+    assertEquals(0, diagnosisKeyRepository.count());
+
+    testRestTemplate.postForEntity("/version/v1/diagnosis-keys", new HttpEntity<>(validSubmissionPayload, headers), Void.class);
+    assertEquals(100, diagnosisKeyRepository.count());
```
Comment (FelixRottler) [inline services/submission/src/test/java/app/coronawarn/server/services/submission/integration/SubmissionPersistenceIT.java:123]:
```suggestion
final int expectedNumberOfKeys =
        validSubmissionPayload.getKeysList().size() * submissionServiceConfig.getRandomKeyPaddingMultiplier();
    assertEquals(expectedNumberOfKeys,
        diagnosisKeyRepository.count());
```

File: common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java [hunk 6]
```diff
@@ -18,12 +19,12 @@
 @DataJdbcTest
 public class DiagnosisKeyRepositoryTest {
 
-  @Autowired
-  private DiagnosisKeyService service;
-
   @Autowired
   private DiagnosisKeyRepository repository;
 
+  private byte[] id = new byte[16];
```
Comment (FelixRottler) [inline common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java:25]: Maybe we can move this variable into the respective tests?

Comment (AliceBilc) [reply common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java:25]: It is used in both of the tests, that's why i thought to have it as a global variable

File: common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java (hunk 2 above)
Comment (AliceBilc) [reply common/persistence/src/test/java/app/coronawarn/server/common/persistence/repository/DiagnosisKeyRepositoryTest.java:63]: Yes, I wanted to check if given a rolling period ourside the range 1-144 -> i have nothing written in the db

**Review by FelixRottler — APPROVED** [review]

**Review by ioangut — APPROVED** [review]

**Review by hilmarf — APPROVED** [review]

**Pull request merged**